lib/puppet/util/windows/sid.rb

```diff
@@ -97,6 +97,29 @@ def octet_string_to_sid_object(bytes)
     end
     module_function :octet_string_to_sid_object
 
+    def ads_to_sid_object(ads_object)
+      if !ads_object || !ads_object.respond_to?(:ole_respond_to?) ||
+        !ads_object.ole_respond_to?(:objectSID) || !ads_object.ole_respond_to?(:Name)
```

Comment:
I'm not totally in love with the duck typing checks here, but at least this makes it easier to do stub objects than checking for `ads_object.is_a?(WIN32OLE)`